src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Conversion.cs

```diff
@@ -362,6 +362,13 @@ private static bool IsFloatingPointExpressionOfUnknownPrecision(BoundExpression
                         explicitCastInCode: explicitCastInCode,
                         rewrittenType: (NamedTypeSymbol)rewrittenType);
 
+                case ConversionKind.MethodGroup when oldNodeOpt is { Type: { TypeKind: TypeKind.FunctionPointer } funcPtrType }:
```

Comment:
>case ConversionKind.MethodGroup when oldNodeOpt is { Type: { TypeKind: TypeKind.FunctionPointer } funcPtrType }: [](start = 16, length = 112)

Should we intercept this node in DiagnosticsPass to prevent its appearance in an expression tree? Please make sure we have a test for an expression tree.